### PR TITLE
Create a new constructor for RoutingOption

### PR DESCRIPTION
**Pull request by matheus220**
The new constructor receives a single input and output port (Character) instead of a Set.

Commit: Add

TaskNumber: #42

### CODE REVIEW

File: src/rbr/RoutingOption.java [hunk 1]
```diff
@@ -13,7 +13,13 @@
         this.ip = ip;
         this.op = op;
     }
-
+    
+    RoutingOption(Character ip, Vertex dst, Character op) {
+        this.dst = dst;
+        this.ip.add(ip);
```
Comment (jmferreiratech) [inline src/rbr/RoutingOption.java:21]: Before using ip/op you need to instantiate them.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]
Use your new constructor at https://github.com/LaNoC-UFC/rsbr/blob/develop/src/rbr/RBR.java#L79

File: src/rbr/RBR.java [hunk 2]
```diff
@@ -66,15 +66,15 @@ public void addRoutingOptions(ArrayList<ArrayList<Path>> paths) {
             for (Path path : alp) {
                 for (Vertex sw : path) {
                     if (path.indexOf(sw) != path.size() - 1) {
-                        Set<Character> op = new HashSet<>();
-                        Set<Character> ip = new HashSet<>();
+                        Character op;
+                        Character ip;
 
-                        op.add(graph.adjunct(sw, path.get(path.indexOf(sw) + 1)).color());
+                        op = graph.adjunct(sw, path.get(path.indexOf(sw) + 1)).color();
```
Comment (jmferreiratech) [inline src/rbr/RBR.java]: Put 'op' declaration together with the assignment.

File: src/rbr/RoutingOption.java [hunk 3]
```diff
@@ -13,7 +13,15 @@
         this.ip = ip;
         this.op = op;
     }
-
+
```
Comment (jmferreiratech) [inline src/rbr/RoutingOption.java]: Remove the trailing spaces.

File: src/rbr/RoutingOption.java [hunk 4]
```diff
@@ -13,7 +13,15 @@
...
-
+    
+    RoutingOption(Character ip, Vertex dst, Character op) {
+        this.dst = dst;
+        this.ip = new HashSet<>();
+        this.op = new HashSet<>();
+        this.ip.add(ip);
+        this.op.add(op);
+    }
+
```
Comment (jmferreiratech) [inline src/rbr/RoutingOption.java]: Remove the trailing spaces.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

**Review by jmferreiratech — APPROVED** [review]

**Review by rmilfont — APPROVED** [review]

**Pull request merged**